cachi2/core/package_managers/gomod.py

```diff
@@ -1151,46 +1149,6 @@ def __exit__(
             super().__exit__(exc, value, tb)
 
 
-def _should_vendor_deps(
-    flags: Iterable[str], app_dir: RootedPath, strict: bool
-) -> Tuple[bool, bool]:
-    """
-    Determine if Cachi2 should vendor dependencies and if it is allowed to make changes.
-
-    This is based on the presence of flags:
-    - gomod-vendor-check => should vendor, can only make changes if vendor dir does not exist
-    - gomod-vendor => should vendor, can make changes
-
-    :param flags: flags from the Cachi2 request
-    :param app_dir: absolute path to the app directory
-    :param strict: fail the request if the vendor dir is present but the flags are not used?
-    :return: (should vendor: bool, allowed to make changes in the vendor directory: bool)
-    :raise PackageRejected: if the vendor dir is present, the flags are not used and we are strict
-    """
-    vendor = app_dir.join_within_root("vendor").path
-
-    if "gomod-vendor-check" in flags:
-        return True, not vendor.exists()
-    if "gomod-vendor" in flags:
-        return True, True
-
-    if strict and vendor.is_dir():
```

Comment:
`strict` is determined by the `gomod_strict_vendor` config option which defaults to True. Deleting this function means we're not taking any action upon setting the variable to False. Now, having the vendor directory and not wanting to vendor it is a weird use case, i.e. `gomod_strict_vendor=False` (we've talked about it), however, you'll still need to deprecate the usage of the variable with a warning in the code (the way our config currently works it's going to be interesting to detect if the user set it somehow) as well as a note in the README.

Reply:
Good catch. I added a deprecation warning when this config option is set, and also updated the corresponding documentation.